backend/src/connection.rs

```diff
@@ -372,22 +396,24 @@ impl Connection {
     /// Creates an encrypted key frame request to be sent to
     /// Connection::outgoing_addr().
     /// Will return None if SRTCP encryption fails.
-    // TODO: Use Result instead of Option
     pub fn send_key_frame_request(
         &mut self,
         key_frame_request: rtp::KeyFrameRequest,
         // It would make more sense to return Option<Packet>, since the outgoing address is fixed,
         // but that actually makes it more difficult for sfu.rs to aggregate the
         // results of calling this across many connections.
         // So we use (packet, addr) for convenience.
-    ) -> Option<(PacketToSend, SocketAddr)> {
-        let outgoing_addr = self.outgoing_addr?;
+    ) -> Result<(PacketToSend, SocketAddr), ConnectionError> {
+        let outgoing_addr = self
+            .outgoing_addr
+            .ok_or(ConnectionError::InvalidOutgoingAddress)?;
         let rtp_endpoint = &mut self.rtp.endpoint;
-        let rtcp_packet = rtp_endpoint.send_pli(key_frame_request.ssrc)?;
-        Some((rtcp_packet, outgoing_addr))
+        let rtcp_packet = rtp_endpoint
+            .send_pli(key_frame_request.ssrc)
+            .map_err(|e| ConnectionError::RtpError(e.to_string()))?;
```

Comment:
I think it might make more sense to just have RtpError be parameterless? Current usage is really only checking for success, and error details aren't noted.

Reply:
Sure thanks for that, I'll refactor accordingly.